graphs/src/anemoi/graphs/processors/post_process.py

```diff
@@ -131,32 +149,95 @@ class RemoveUnconnectedNodes(BaseNodeMaskingProcessor):
 
     def __init__(
         self,
-        nodes_name: str,
+        nodes_name: Union[str, Iterable[str]],
         save_mask_indices_to_attr: str | None = None,
         ignore: str | None = None,
     ) -> None:
         super().__init__(nodes_name, save_mask_indices_to_attr)
         self.ignore = ignore
 
-    def compute_mask(self, graph: HeteroData) -> torch.Tensor:
+    def compute_mask(self, graph: HeteroData, nodes_name: str) -> torch.Tensor:
         """Compute the mask of connected nodes."""
-        nodes = graph[self.nodes_name]
+        nodes = graph[nodes_name]
         connected_mask = torch.zeros(nodes.num_nodes, dtype=torch.bool)
 
         if self.ignore is not None:
             LOGGER.info(f"The nodes with {self.ignore}=True will not be removed.")
             connected_mask[nodes[self.ignore].bool().squeeze()] = True
 
         for (source_name, _, target_name), edges in graph.edge_items():
-            if source_name == self.nodes_name:
+            if source_name == nodes_name:
                 connected_mask[edges.edge_index[0]] = True
 
-            if target_name == self.nodes_name:
+            if target_name == nodes_name:
                 connected_mask[edges.edge_index[1]] = True
 
         return connected_mask
 
 
+class SubsetNodesInArea(BaseNodeMaskingProcessor):
```

Comment:
could be have a test for this also on `graphs/tests/processors/test_post_process.py`?